src/interface/src/app/map.service.ts

```diff
@@ -39,17 +39,34 @@ export class MapService {
     );
   }
 
-  getHUC12BoundaryShapes(): Observable<GeoJSON.GeoJSON> {
+  regionToString(region: Region): string {
+    switch (region) {
+      case Region.SIERRA_NEVADA: return "SierraNevada";
+      case Region.CENTRAL_COAST: return "CentralCoast";
+      case Region.NORTHERN_CALIFORNIA: return "NorthernCalifornia";
+      case Region.SOUTHERN_CALIFORNIA: return "SouthernCalifornia";
+    }
+  }
+
+  getHUC12BoundaryShapes(region: Region|null): Observable<GeoJSON.GeoJSON> {
     // Get the shapes from the REST server.
+    var regionString: string = '';
+    if (region != null) {
```

Comment:
What does the HTTP request return if the region param is empty?

Reply:
It will return all the boundaries, and not clip them.